executor/metrics_reader.go

```diff
@@ -120,17 +118,64 @@ func (e *MetricRetriever) queryMetric(ctx context.Context, sctx sessionctx.Conte
 	return result, err
 }
 
+type metricStorage struct {
+	PdServer struct {
+		MetricStorage string `json:"metric-storage"`
+	} `json:"pd-server"`
+}
+
+type prometheus struct {
+	IP         string `json:"ip"`
+	BinaryPath string `json:"binary_path"`
+	Port       int    `json:"port"`
+}
+
 func (e *MetricRetriever) getMetricAddr(sctx sessionctx.Context) (string, error) {
 	// Get PD servers info.
 	store := sctx.GetStore()
 	etcd, ok := store.(tikv.EtcdBackend)
 	if !ok {
 		return "", errors.Errorf("%T not an etcd backend", store)
 	}
-	for _, addr := range etcd.EtcdAddrs() {
-		return addr, nil
+
+	pdAddrs := etcd.EtcdAddrs()
+	if len(pdAddrs) < 0 {
+		return "", errors.Errorf("pd unavailable")
+	}
+	var res string
+
+	// get prometheus address from pdApi
```

Comment:
redundant code with info?

Reply:
Yes, this func is redundant, i forgot this place, deleted it.